clap_derive/tests/arg_enum.rs

```diff
@@ -0,0 +1,189 @@
+fn basic() {
+    #[derive(Clap, PartialEq, Debug)]
+    enum ArgChoice {
+        Foo,
+        Bar,
+    }
+
+    #[derive(Clap, PartialEq, Debug)]
+    struct Opt {
+        #[clap(arg_enum)]
+        arg: ArgChoice,
+    };
+
+    assert_eq!(
+        Opt {
+            arg: ArgChoice::Foo
+        },
+        Opt::parse_from(&["", "foo"])
+    );
+    assert_eq!(
+        Opt {
+            arg: ArgChoice::Bar
+        },
+        Opt::parse_from(&["", "bar"])
+    );
+    assert!(Opt::try_parse_from(&["", "fOo"]).is_err());
+}
+
+#[test]
+fn multi_word_is_renamed_kebab() {
+    #[derive(Clap, PartialEq, Debug)]
+    #[allow(non_camel_case_types)]
+    enum ArgChoice {
+        FooBar,
+        BAR_BAZ,
+    }
+
+    #[derive(Clap, PartialEq, Debug)]
+    struct Opt {
+        #[clap(arg_enum)]
+        arg: ArgChoice,
+    };
+
+    assert_eq!(
+        Opt {
+            arg: ArgChoice::FooBar
+        },
+        Opt::parse_from(&["", "foo-bar"])
+    );
+    assert_eq!(
+        Opt {
+            arg: ArgChoice::BAR_BAZ
+        },
+        Opt::parse_from(&["", "bar-baz"])
+    );
+    assert!(Opt::try_parse_from(&["", "FooBar"]).is_err());
+}
+
+#[test]
+fn variant_with_defined_casing() {
+    #[derive(Clap, PartialEq, Debug)]
+    enum ArgChoice {
+        #[clap(rename_all = "screaming_snake")]
+        FooBar,
+    }
+
+    #[derive(Clap, PartialEq, Debug)]
+    struct Opt {
+        #[clap(arg_enum)]
+        arg: ArgChoice,
+    };
+
+    assert_eq!(
+        Opt {
+            arg: ArgChoice::FooBar
+        },
+        Opt::parse_from(&["", "FOO_BAR"])
+    );
+    assert!(Opt::try_parse_from(&["", "FooBar"]).is_err());
+}
+
+#[test]
+fn casing_is_propogated_from_parent() {
+    #[derive(Clap, PartialEq, Debug)]
+    #[clap(rename_all = "screaming_snake")]
+    enum ArgChoice {
+        FooBar,
+    }
+
+    #[derive(Clap, PartialEq, Debug)]
+    struct Opt {
+        #[clap(arg_enum)]
+        arg: ArgChoice,
+    };
+
+    assert_eq!(
+        Opt {
+            arg: ArgChoice::FooBar
+        },
+        Opt::parse_from(&["", "FOO_BAR"])
+    );
+    assert!(Opt::try_parse_from(&["", "FooBar"]).is_err());
+}
+
+#[test]
+fn casing_propogation_is_overridden() {
+    #[derive(Clap, PartialEq, Debug)]
+    #[clap(rename_all = "screaming_snake")]
+    enum ArgChoice {
+        #[clap(rename_all = "camel")]
+        FooBar,
+    }
+
+    #[derive(Clap, PartialEq, Debug)]
+    struct Opt {
+        #[clap(arg_enum)]
+        arg: ArgChoice,
+    };
+
+    assert_eq!(
+        Opt {
+            arg: ArgChoice::FooBar
+        },
+        Opt::parse_from(&["", "fooBar"])
+    );
+    assert!(Opt::try_parse_from(&["", "FooBar"]).is_err());
+    assert!(Opt::try_parse_from(&["", "FOO_BAR"]).is_err());
+}
+
+#[test]
+fn case_insensitive() {
+    #[derive(Clap, PartialEq, Debug)]
+    enum ArgChoice {
+        Foo,
+    }
+
+    #[derive(Clap, PartialEq, Debug)]
+    struct Opt {
+        #[clap(arg_enum, case_insensitive(true))]
```

Comment:
I think `#[clap(arg_enum(case_insensitive))]` syntax would look better, don't you think? `true` if present, `false` if omitted.

Reply:
The `case_insensitive` is being used by `arg`. I just read that usage. We don't need anything special here.